pkg/fanal/analyzer/sbom/sbom.go

```diff
@@ -58,10 +57,14 @@ func (a sbomAnalyzer) Analyze(ctx context.Context, input analyzer.AnalysisInput)
 		bom.Packages[i].FilePath = path.Join(input.FilePath, pkgInfo.FilePath)
 	}
 
-	// FilePath for apps with aggregatingTypes is empty.
-	// Set the SBOM file path as Application.FilePath to correctly overwrite applications when merging layers.
+	// There are cases when FilePath for Application is empty:
+	// - FilePath for apps with aggregatingTypes is empty.
+	// - Third party SBOM without Application component.
+	// We need to use SBOM file path as Application.FilePath to correctly:
+	// - overwrite applications when merging layers. See https://github.com/aquasecurity/trivy/issues/7851
+	// - show FilePath as Target of Application. See https://github.com/aquasecurity/trivy/issues/8189.
 	for i, app := range bom.Applications {
-		if slices.Contains(ftypes.AggregatingTypes, app.Type) && app.FilePath == "" {
+		if app.FilePath == "" {
 			bom.Applications[i].FilePath = input.FilePath
```

Comment:
I didn't add the type prefix (as we discussed in https://github.com/aquasecurity/trivy/discussions/7556#discussioncomment-12038735) because `name` is not a unique field (for CycloneDX and SPDX), so we can use the same name for multiple components.